cmd/watch-termination/main.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+)
+
+func main() {
+	terminationLog := flag.String("termination-log-file", "", "Write logs after SIGTERM to this file (in addition to stderr)")
+	terminationLock := flag.String("termination-touch-file", "", "Touch this file on SIGTERM and delete on termination")
+
+	flag.Parse()
+	args := flag.CommandLine.Args()
+
+	if len(args) == 0 {
+		fmt.Println("Missing command line")
+		os.Exit(1)
+	}
+
+	// use special tee-like writer when termination log is set
+	termCh := make(chan struct{})
+	var stderr io.Writer = os.Stderr
+	if len(*terminationLog) > 0 {
+		stderr = &terminationFileWriter{
+			Writer:             os.Stderr,
+			fn:                 *terminationLog,
+			startFileLoggingCh: termCh,
+		}
+	}
+
+	cmd := exec.Command(args[0], args[1:]...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = stderr
+
+	// forward SIGTERM and SIGINT to child
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGTERM, syscall.SIGINT)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for s := range sigCh {
+			fmt.Fprintf(stderr, "Received signal %s\n", s)
+
+			if len(*terminationLock) > 0 {
+				fmt.Fprintln(stderr, "Touching", *terminationLock)
+				if err := touch(*terminationLock); err != nil {
+					fmt.Fprintln(stderr, fmt.Errorf("error touching %s: %v", *terminationLock, err))
+					// keep going
+				}
+			}
+
+			select {
+			case <-termCh:
+			default:
+				close(termCh)
+			}
+			cmd.Process.Signal(s)
+		}
+	}()
+
+	fmt.Printf("Launching %v\n", cmd)
+	rc := 0
+	if err := cmd.Run(); err != nil {
+		if exitError, ok := err.(*exec.ExitError); !ok {
+			rc = exitError.ExitCode()
+		} else {
+			log.Fatal(err)
```

Comment:
why do we want to crash here? (I think that this will `os.Exit(255)`)

Reply:
also fine. It's an error we will probably never see.

Reply:
fixed